thrall/app/lib/kinesis/ThrallEventConsumer.scala

```diff
@@ -74,8 +78,10 @@ class ThrallEventConsumer(es: ElasticSearch6,
   }
 
   private def processUpdateMessage(updateMessage: UpdateMessage) = {
+    implicit val mc: MarkerContext = updateMessage.toLogMarker
+
     messageProcessor.chooseProcessor(updateMessage).map { messageProcessor =>
-      Try(Await.ready(messageProcessor.apply(updateMessage), Timeout))
+      Try(Await.ready(RetryHandler.handleWithRetryAndTimeout(() => messageProcessor.apply(updateMessage),2, Timeout).apply(), Timeout))
```

Comment:
This is a chunky nugget of code! Would it be possible to split it up to make it more readable? 😁 🍗

Reply:
I've pulled it over some lines and commented them, not sure how much we can split them